### ISSUE TITLE
Using 2 Conv Layer

### ISSUE THREAD

**Issue by ShoaibSajid** [vzhou842/cnn-from-scratch#3]
Hello.

Thank for such an amazing code. It has helped me a lot in understanding.

I am trying to add another Conv layer.
I was able to change conv to a 3d conv layer. 
The forward propagation works fine. 
but in backward propagation, I am unable to pass derivatives from one conv layer to the other.

Is it possible for you to show me how I can calculate dL/dInput for the conv layer. (the gradient which will be passed from second conv layer to the first conv layer during backpropagation)

**Comment by zitongbai** [comment]
> Hello.
> 
> Thank for such an amazing code. It has helped me a lot in understanding.
> 
> I am trying to add another Conv layer. I was able to change conv to a 3d conv layer. The forward propagation works fine. but in backward propagation, I am unable to pass derivatives from one conv layer to the other.
> 
> Is it possible for you to show me how I can calculate dL/dInput for the conv layer. (the gradient which will be passed from second conv layer to the first conv layer during backpropagation)

You may refer to [my fork](https://github.com/zitongbai/cnn-from-scratch), where I have implemented multi-channel input for Conv3x3 and refined the computation of backward propagation. This enhancement enables the realization of multi-layer CNN neural networks.
